### PR TITLE
Adapt pod killing and cleanup for generic container runtime

### PR DESCRIPTION
**Pull request by yujuhong**
This change removes docker-specifc code in killUnwantedPods. It
also instructs the cleanup code to move away from interacting with
containers directly. They should always deal with the pod-level
abstraction if at all possible.

### CODE REVIEW

**Comment by yujuhong** [comment]
/cc @vmarmol, @yifan-gu 

To avoid inspecting individual containers, I had to use `containerManager.GetPods`. This bypasses the runtime cache and may affect performance (?). I am doubtful that this operation is so expensive that we need to use the runtime cache at all times (since this is called only once per sync). But I suppose the cache was added for a reason.... If we absolutely want to avoid calling this function, I could modify the container runtime interface to support a function similar to `GetRunningContainers` to take container ids, inspect them, and return running pods. This doesn't seem to play well with rkt though.

**Comment by vmarmol** [comment]
LGTM. Nice! We take out a bunch of code and end up with something more generic :)

I agree that the performance impact is probably minimal since this is only a single extra call per sync loop. If anything, with time this operation will become async too.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1387,16 +1381,24 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
...
+	err = kl.killUnwantedPods(desiredPods, runningPods)
 	if err != nil {
 		glog.Errorf("Failed killing unwanted containers: %v", err)
+	}
+
+	// Note that we just killed the unwanted pods. This may not have reflected
+	// in the cache. We need to bypass the cach to get the latest set of
+	// running pods to clean up the volumes.
+	// TODO: Evaluate the performance impact of bypassing the runtime cache.
+	runningPods, err = kl.containerManager.GetPods(true)
```
Comment (yifan-gu) [inline pkg/kubelet/kubelet.go]: Why we pass `true` here? I remember `true` is for getting all pods including dead ones?

**Comment by yifan-gu** [comment]
Great! This is what I want to do! Thanks @yujuhong !

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: Good catch. I meant to say false...Thanks!

**Comment by yujuhong** [comment]
Rebased. Thanks!

**Comment by yifan-gu** [comment]
LGTM

**Pull request merged**